### PR TITLE
1-4 tasks - Egor Golubev

### PR DESCRIPTION
**Pull request by EgorHolubeu**
https://travis-ci.com/github/EgorHolubeu/js-assignments/builds/174611845

### CODE REVIEW

File: task/02-numbers-tasks.js [hunk 1]
```diff
@@ -221,7 +231,7 @@ function isPrime(n) {
  *   toNumber(new Number(42), 0) => 42
  */
 function toNumber(value, def) {
-    throw new Error('Not implemented');
+    return +value ? +value : def;
 }
```
Comment (FedorovD) [inline task/02-numbers-tasks.js:235]: not correct, what if the value is '0'

File: task/03-date-tasks.js [hunk 2]
```diff
@@ -76,14 +78,15 @@ function isLeapYear(date) {
  *    Date(2000,1,1,10,0,0),  Date(2000,1,1,15,20,10,453)   => "05:20:10.453"
  */
 function timeSpanToString(startDate, endDate) {
-   throw new Error('Not implemented');
+    let time = new Date(endDate - startDate);
+    return `${parseInt((endDate.getTime() - startDate.getTime()) / (1000 * 60 * 60)) < 10 ? '0' + parseInt((endDate.getTime() - startDate.getTime()) / (1000 * 60 * 60)) : parseInt((endDate.getTime() - startDate.getTime()) / (1000 * 60 * 60))}:${time.getMinutes() < 10 ? '0' + time.getMinutes() : time.getMinutes()}:${time.getSeconds() < 10 ? '0' + time.getSeconds() : time.getSeconds()}.${time.getMilliseconds() < 10 ? '00' + time.getMilliseconds() : time.getMilliseconds() < 100 ? '0' + time.getMilliseconds() : time.getMilliseconds()}`;
 }
```
Comment (FedorovD) [inline task/03-date-tasks.js:109]: avoid this code style, if-else is more readable than multiple conditional operators ? :

File: task/04-arrays-tasks.js [hunk 3]
```diff
@@ -286,16 +292,22 @@ function propagateItemsByPositionIndex(arr) {
...
 function get3TopItems(arr) {
-   throw new Error('Not implemented');
+    function compare(a, b) {
+        if (a > b) return -1;
+        if (a === b) return 0;
+        if (a < b) return 1;
+    }
+
+    return arr.sort(compare).splice(0, 3);
 }
```
Comment (FedorovD) [inline task/04-arrays-tasks.js:300]: there is no reason to create an extra function, you can define it right in the `sort` call
.sort((a, b) => {...})

File: task/04-arrays-tasks.js [hunk 4]
```diff
@@ -415,61 +428,81 @@ function toStringList(arr) {
...
+        if (a.country === b.country) {
+            if (a.city > b.city) return 1;
+            if (a.city === b.city) return 0;
+            if (a.city < b.city) return -1;
+        }
+        if (a.country < b.country) return -1;
+    }
+
+    return arr.sort(compare);
 }
```
Comment (FedorovD) [inline task/04-arrays-tasks.js:438]: same as above

**Comment by FedorovD** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by EgorHolubeu** [comment]
https://travis-ci.com/github/EgorHolubeu/js-assignments/builds/175772136

**Comment by FedorovD** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**